schema/proto3/goalstateprovisioner.proto

```diff
@@ -32,30 +32,57 @@ service GoalStateProvisioner {
     //
     // Note: It is a NoOps for Control Agents when the operation type is INFO or GET.
     //       Use RetrieveNetworkResourceStates for state query.
+    // this is for DPM->ACA
     rpc PushNetworkResourceStates (GoalState) returns (GoalStateOperationReply) {
     }
 
-    // same as PushNetworkResourceStates with streaming GoalState and streaming GoalStateOperationReply
-    rpc PushNetworkResourceStatesStream (stream GoalState) returns (stream GoalStateOperationReply) {
+    // similar to PushNetworkResourceStates with streaming GoalStateV2 and streaming GoalStateOperationReply
+    // this is for DPM->NCM, and NCM->ACA
+    rpc PushNetworkResourceStatesStream (stream GoalStateV2) returns (stream GoalStateOperationReply) {
     }
 
-
-    // Retrieve a group of network resource states (stored as a steam of GoalState objects)
-    rpc RetrieveNetworkResourceStates (GoalStateRequest) returns (stream GoalState) {
+    // Request network configuration for ACA on-demand processing and also restart handling
+    //
+    // Input: a NetworkConfigurationsRequest object consists of a list of operation requests, and each request contains an operation type and a resource configuration
+    // Results consist of a list of operation statuses, and each status is a response to one operation request in the input
+    // this is for ACA->NCM
+    rpc RequestNetworkConfigurations (NetworkConfigurationsRequest) returns (NCMReply) {
     }
+
 }
 
-message GoalStateRequest {
+message NetworkConfigurationsRequest {
     uint32 format_version = 1;
 
     message ResourceStateRequest {
-        string resource_id = 1;
-        ResourceType resource_type = 2;
+        RequestType request_type = 1;
+        string request_id = 2; // UUID generated by ACA
+        uint32 tunnel_id = 3; 
+        string source_ip = 4;
+        uint32 source_port = 5;
+        string destination_ip = 6; 
+        uint32 destination_port = 7;
+        EtherType ethertype = 8;
+        Protocol protocol = 9;
     }
 
     repeated ResourceStateRequest state_requests = 2;
 }
 
+message NCMReply {
```

Comment:
Let us use some more generic name, instead of a name tied to a component, e.g., OnDemandRequestReply.